android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java

```diff
@@ -88,99 +99,176 @@ protected void onLayout(boolean changed, int left, int top, int right, int botto
-
-			                Layout layout = textView.getLayout();
-			                int line = layout.getLineForVertical(y);
-			                int off = layout.getOffsetForHorizontal(line, x);
-
-			                ClickableSpan[] link = buffer.getSpans(off, off,
-			                        ClickableSpan.class);
-
-							if (link.length != 0) {
-								ClickableSpan cSpan = link[0]; 
-								if (action == MotionEvent.ACTION_UP) {
-									TiViewProxy proxy = getProxy();
-									if(proxy.hasListeners("link") && (cSpan instanceof URLSpan)) {
-										KrollDict evnt = new KrollDict();
-										evnt.put("url", ((URLSpan)cSpan).getURL());
-										proxy.fireEvent("link", evnt, false);
-									} else {
-										cSpan.onClick(textView);
-									}
-								} else if (action == MotionEvent.ACTION_DOWN) {
-									Selection.setSelection(buffer, buffer.getSpanStart(cSpan), buffer.getSpanEnd(cSpan));
+				TextView textView = (TextView) this;
+				Object text = textView.getText();
+
+				//For html texts, we will manually detect url clicks.
+				if (text instanceof SpannedString) {
+					SpannedString spanned = (SpannedString) text;
+					Spannable buffer = Factory.getInstance().newSpannable(spanned.subSequence(0, spanned.length()));
+
+					int action = event.getAction();
+
+					if (action == MotionEvent.ACTION_UP || action == MotionEvent.ACTION_DOWN) {
+						int x = (int) event.getX();
+						int y = (int) event.getY();
+
+						x -= textView.getTotalPaddingLeft();
+						y -= textView.getTotalPaddingTop();
+
+						x += textView.getScrollX();
+						y += textView.getScrollY();
+
+						Layout layout = textView.getLayout();
+						int line = layout.getLineForVertical(y);
+						int off = layout.getOffsetForHorizontal(line, x);
+
+						ClickableSpan[] link = buffer.getSpans(off, off, ClickableSpan.class);
+
+						if (link.length != 0) {
+							ClickableSpan cSpan = link[0]; 
+							if (action == MotionEvent.ACTION_UP) {
+								TiViewProxy proxy = getProxy();
+								if(proxy.hasListeners("link") && (cSpan instanceof URLSpan)) {
+									KrollDict evnt = new KrollDict();
+									evnt.put("url", ((URLSpan)cSpan).getURL());
+									proxy.fireEvent("link", evnt, false);
+								} else {
+									cSpan.onClick(textView);
 								}
+							} else if (action == MotionEvent.ACTION_DOWN) {
+								Selection.setSelection(buffer, buffer.getSpanStart(cSpan), buffer.getSpanEnd(cSpan));
 							}
-			            }
+						}
+					}
 
-			        }
-			        return super.onTouchEvent(event);
-			    }
+				}
+				return super.onTouchEvent(event);
+			}
 		};
 		tv.setGravity(Gravity.CENTER_VERTICAL | Gravity.LEFT);
 		tv.setPadding(0, 0, 0, 0);
-		tv.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_FLAG_MULTI_LINE);
-		tv.setKeyListener(null);
 		tv.setFocusable(false);
-		tv.setSingleLine(false);
-		tv.setEllipsize(ellipsize);
+		tv.setEllipsize(this.ellipsize);
+		tv.setSingleLine(!this.wordWrap);
```

Comment:
keep this as:
```Java
tv.setSingleLine(false);
```
as this will prevent the use of multi-lined labels (see [TIMOB-24851](https://jira.appcelerator.org/browse/TIMOB-24851)